jinja2/includes/google_analytics.html

```diff
@@ -1,65 +1,67 @@
 {% if config.GOOGLE_ANALYTICS_ACCOUNT != '0' %}
 {% set GA_FILE = 'analytics_debug' if settings.DEBUG else 'analytics' %}
 <script type="text/javascript">
+    // only load GA if DNT is not enabled
+    if (Mozilla && !Mozilla.dntEnabled()) {
+        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+        })(window,document,'script','//www.google-analytics.com/{{ GA_FILE }}.js','ga');
```

Comment:
This bit of compressed JS does a couple of things. It create a global object ``ga`` that stores whatever you call on it, which then gets used in the script. It also adds a ``<script>`` tag that loads Google Analytics. When that loads, it creates the "real" ``ga`` object and sends the cached calls to Google. It's a clever bit of JS.

Here's my "uncompressed" version:

```js
(function(win, doc, tagName, src, ga, elem, elemCreated) {
    win['GoogleAnalyticsObject'] = ga;
    win[ga] = win[ga] || function() {
        (win[ga].q = win[ga].q || []).push(arguments)
    }, win[ga].l = 1 * new Date();
    elem = doc.createElement(tagName);
    elemCreated = doc.getElementsByTagName(tagName)[0];
    elem.async = 1;
    elem.src = src;
    elemCreated.parentNode.insertBefore(elem, elemCreated)
})(window, document, 'script', '//www.google-analytics.com/{{ GA_FILE }}.js', 'ga');
```

We could split this up: 

1. Create the global caching ``ga`` object, and populate it no matter what.  Something like:

```js
(function(win, ga, src) {
    win['GoogleAnalyticsObject'] = ga;
    win[ga] = win[ga] || function() {
        (win[ga].q = win[ga].q || []).push(arguments)
    }, win[ga].l = 1 * new Date();
    win['GoogleAnalyticsSrc'] = src;
})(window, '//www.google-analytics.com/{{ GA_FILE }}.js', 'ga');
```
2. Run the DNT code, and only if DNT is not enabled, add the ``<script>`` to load the real GA code. We can use pipeline to combine the upstream DNT code with the enabling code:

```
if !Mozilla.dntEnabled() {
    (function(win, srcName, doc, tagName, elem, elemCreated) {
        elem = doc.createElement(tagName),
        elemCreated = doc.getElementsByTagName(tagName)[0];
        elem.async = 1;
        elem.src = win[srcName];
        elemCreated.parentNode.insertBefore(elem, elemCreated)
    })(window, 'GoogleAnalyticsSrc', document, 'script');
}
```

I suspect that if I read the GA code, it also does DNT detection. But I'm at my limit for JavaScript for the morning.

Reply:
Hmmm so, I have two concerns here:

1. Do we really want to mess with Google's code? What you are suggesting seems sound but,
2. If GA then is not loaded all calls to `ga` inside the code will actually be execute and pushed onto the tmp array

Depending on the number of calls on a single page this is more than likely not a big deal but, I am not sure its required. The concern Stephanie had is resolved by moving the analytics block to below the CSS block, and moving the GA calls in `task-completion` to `analytics.js`

Thoughts @jwhitlock @stephaniehobson

Reply:
If you are both happy with the current solution, then we should ship it.

Reply:
Sure thing, lets wait and see what Stephanie thinks. Thanks!

Reply:
Yeah, I'd be nervous about messing with the ga code. Scrapers and stuff get annoyed at us if we mess with it (it can break our webmaster tools integration for example). 

I wish we could though, it would be nice to push the date stamp earlier in the rendering process.